gh-actions/prepare-host/action.yaml

```diff
@@ -0,0 +1,24 @@
+name: 'Prepare host'
```

Comment:
Generally this is only ever used for E2E, can we just fold it into the "Run E2E" shared action?

Reply:
What do you mean, wait...

Reply:
We only need to "prepare host" before we run the E2E.
I think this can all be in a shared "run E2E" shared action (and if necessary we can split it out in the future though I doubt it will be).